test/test_tensorexpr_pybind.py

```diff
@@ -59,5 +66,34 @@ def test_external_calls(self):
             codegen.call([tA, tB, tC])
             torch.testing.assert_allclose(torch.matmul(tA, tB), tC)
 
+
+    @unittest.skipIf(not LLVM_ENABLED, "LLVM backend not enabled")
+    def test_kernel(self):
+        def f(a, b, c):
+            return a + b + c
+        device, size = 'cpu', (4, 4)
+        x = torch.rand(size, device=device)
+        y = torch.rand(size, device=device)
+        z = torch.rand(size, device=device)
+
+        scripted_f = torch.jit.script(f, (x, y, z))
+
+        scripted_f(x, y, z)
+        scripted_f(x, y, z)
+
+        node = None
+        graph = torch.jit.last_executed_optimized_graph()
+        node = graph.findNode("prim::TensorExprGroup", True)
```

Comment:
Instead of going through the fuser pass I would rather construct a graph here directly - it also would probably be more compact too. Dependence on the fuser is unnecessary and could accidentally result in a failure for this test should the fuser somehow change its behavior.

You could find an example of how to construct a graph from a string here:
https://github.com/pytorch/pytorch/blob/2df4acd0253f32a91b9a11a4b1e7c8d392a17a35/test/test_jit.py#L1316-L1334

You could specify the necessary shape information there as well.